src/SW_Flow_lib.c

```diff
@@ -1712,17 +1742,18 @@ Based on equations from Eitzinger 2000. @cite Eitzinger2000
 
 unsigned int adjust_Tsoil_by_freezing_and_thawing(double oldavgLyrTemp[], double avgLyrTemp[],
 	double shParam, unsigned int nlyrs, double vwc[], double bDensity[]){
-// Calculate fusion pools based on soil profile layers, soil freezing/thawing, and if freezing/thawing not completed during one day, then adjust soil temperature
-// based on Eitzinger, J., W. J. Parton, and M. Hartman. 2000. Improvement and Validation of A Daily Soil Temperature Submodel for Freezing/Thawing Periods. Soil Science 165:525-534.
+	// Calculate fusion pools based on soil profile layers, soil freezing/thawing, and if freezing/thawing not completed during one day, then adjust soil temperature
+	// based on Eitzinger, J., W. J. Parton, and M. Hartman. 2000. Improvement and Validation of A Daily Soil Temperature Submodel for Freezing/Thawing Periods. Soil Science 165:525-534.
 
-// NOTE: THIS FUNCTION IS CURRENTLY NOT OPERATIONAL: DESCRIPTION BY EITZINGER ET AL. 2000 SEEMS INSUFFICIENT
+	// NOTE: THIS FUNCTION IS CURRENTLY NOT OPERATIONAL: DESCRIPTION BY EITZINGER ET AL. 2000 SEEMS INSUFFICIENT
 
-//	double deltaTemp, Cis, sFusionPool[nlyrs], sFusionPool_actual[nlyrs];
-// To avoid compiler warnings "warning: parameter set but not used"
-double temp;
-temp = oldavgLyrTemp[0] + avgLyrTemp[0] + shParam + nlyrs + vwc[0] + bDensity[0];
-temp += temp;
-// end avoid compiler warnings
+	//	double deltaTemp, Cis, sFusionPool[nlyrs], sFusionPool_actual[nlyrs];
+	// To avoid compiler warnings "warning: parameter set but not used"
+	double temp;
+	temp = oldavgLyrTemp[0] + avgLyrTemp[0] + shParam + nlyrs + vwc[0] + bDensity[0];
+	temp += temp;
+	// end avoid compiler warnings
+	ST_RGR_VALUES dummyInstance;

```

Comment:
Why convert the global object `ST_RGR_VALUES *st = &stValues;`  into a local copy `ST_RGR_VALUES dummyInstance;`?

Why not upgrade to a function argument `ST_RGR_VALUES* SW_StRegValues` so that it can be modified (as done elsewhere)?

Reply:
`adjust_Tsoil_by_freezing_and_thawing()`